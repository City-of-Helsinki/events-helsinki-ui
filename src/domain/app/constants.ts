export enum ROUTES {
  ABOUT = '/about',
  COLLECTION = '/collection/:id',
  COLLECTIONS = '/collections',
  EVENTS = '/events',
  EVENT = '/event/:id',
  HOME = '/home',
  HOME_PREVIEW = '/home/:id',
}
