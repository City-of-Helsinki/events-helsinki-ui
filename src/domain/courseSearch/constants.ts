export const DEFAULT_SEARCH_FILTERS = {
  categories: [],
  dateTypes: [],
  divisions: [],
  end: null,
  isFree: false,
  onlyOngoingCourses: false,
  places: [],
  start: null,
  text: [],
};
