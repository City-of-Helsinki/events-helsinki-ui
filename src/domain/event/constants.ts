export const mockEventData = {
  eventDetails: {
    description: {
      en: "description en",
      fi: "description fi",
      sv: "description sv"
    },
    endTime: null,
    externalLinks: [
      {
        link: "https://www.facebook.com/events/12345687/",
        name: "extlink_facebook"
      }
    ],
    id: "123",
    images: [
      {
        id: "60578",
        name: "piha-2.jpg",
        url: "https://api.hel.fi/linkedevents/media/images/piha-2_yYlCdPB.jpg"
      }
    ],
    inLanguage: [
      {
        id: "fi",
        name: {
          en: "Finnish",
          fi: "suomi",
          sv: "finska"
        },
        translationAvailable: true
      }
    ],
    infoUrl: {
      en:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      fi:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      sv:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080"
    },
    keywords: [],
    location: {
      addressLocality: {
        en: "locality en",
        fi: "locality fi",
        sv: "locality sv"
      },
      divisions: [],
      email: "location@info.com",
      infoUrl: {
        en: "http://www.infourl.com",
        fi: "http://www.infourl.com",
        sv: "http://www.infourl.com"
      },
      name: {
        en: "name en",
        fi: "name fi",
        sv: "name sv"
      },
      position: {
        coordinates: [25.01497, 60.250507],
        type: "Point"
      },
      postalCode: "01234",
      streetAddress: {
        en: "streetAddress en",
        fi: "streetAddress fi",
        sv: "streetAddress sv"
      },
      telephone: {
        en: "044 123 4567",
        fi: "044 123 4567",
        sv: "044 123 4567"
      }
    },
    name: {
      en: "name en",
      fi: "name fi",
      sv: "name sv"
    },
    offers: [
      {
        description: null,
        infoUrl: null,
        isFree: true,
        price: null
      }
    ],
    provider: null,
    publisher: "provider:123",
    shortDescription: {
      en: "short description en",
      fi: "short description fi",
      sv: "short description sv"
    },
    startTime: "2019-12-13T08:00:00Z"
  }
};

// Amount of the similar events
export const SIMILAR_EVENTS_AMOUNT = 8;

export const EVENT_KEYWORD_BLACK_LIST = [
  "helmet:10592", // Yleinen
  "helmet:10689", // Helsinki
  "helmet:11916", // Infonäytöt (Helsinki)
  "kulke:53", // Suomi
  "kulke:54", // Ruotsi
  "kulke:55", // Englanti
  "kulke:616", // Näkyy TV-sovelluksessa
  "kulke:671", // Normaali
  "kulke:673" // Suomi (TV)
];
