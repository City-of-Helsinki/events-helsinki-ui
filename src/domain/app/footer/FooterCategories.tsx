import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router';

import CategoryFilter from '../../../common/components/category/CategoryFilter';
import { Category } from '../../../common/types';
import useLocale from '../../../hooks/useLocale';
import scrollToTop from '../../../util/scrollToTop';
import { DEFAULT_SEARCH_FILTERS } from '../../eventSearch/constants';
import { getCategoryOptions, getSearchQuery } from '../../eventSearch/utils';
import { ROUTES } from '../constants';
import styles from './footerCategories.module.scss';

const FooterCategories: FunctionComponent = () => {
  const { t } = useTranslation();
  const locale = useLocale();
  const { push } = useHistory();

  const handleCategoryClick = (category: Category) => {
    const search = getSearchQuery({
      ...DEFAULT_SEARCH_FILTERS,
      categories: [category.value],
    });

    push({ pathname: `/${locale}${ROUTES.EVENTS}`, search });
    scrollToTop();
  };

  const categories = getCategoryOptions(t);

  return (
    <div className={styles.topFooterWrapper}>
      <hr className={styles.divider} aria-hidden />
      <h2 className={styles.categoriesTitle}>{t('footer.titleCategories')}</h2>
      <div className={styles.categoriesInnerWrapper}>
        {categories.map((category) => {
          return (
            <CategoryFilter
              key={category.value}
              hasHorizontalPadding={true}
              icon={category.icon}
              onClick={handleCategoryClick}
              text={category.text}
              value={category.value}
            />
          );
        })}
      </div>
    </div>
  );
};

export default FooterCategories;
