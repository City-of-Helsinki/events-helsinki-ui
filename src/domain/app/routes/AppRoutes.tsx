import React from 'react';
import { useTranslation } from 'react-i18next';
import { Redirect, Route, Switch } from 'react-router';

import { getCurrentLanguage } from '../../../common/translation/TranslationUtils';
import { SUPPORT_LANGUAGES } from '../../../constants';
import { ROUTES } from './constants';
import App from './LocaleRoutes';

const localeParam = `:locale(${SUPPORT_LANGUAGES.EN}|${SUPPORT_LANGUAGES.FI}|${SUPPORT_LANGUAGES.SV})`;

const AppRoutes: React.FC = () => {
  const { i18n } = useTranslation();
  const locale = getCurrentLanguage(i18n);

  return (
    <Switch>
      {/* Redirect '/' straight to '/:locale/home' to avoid multiple redirections */}
      <Redirect exact path="/" to={`/${locale}${ROUTES.HOME}`} />
      <Redirect exact path={`/${locale}`} to={`/${locale}${ROUTES.HOME}`} />
      <Route path={`/${localeParam}`} component={App} />
      <Route
        render={(props) => (
          // Add locale to path if is is missing
          <Redirect to={`/${locale}${props.location.pathname}`} />
        )}
      />
    </Switch>
  );
};

export default AppRoutes;
