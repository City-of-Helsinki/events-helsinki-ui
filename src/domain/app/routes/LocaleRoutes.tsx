import '../../../assets/styles/main.scss';
import 'hds-core/lib/base.css';
import 'hds-core/lib/components/all.css';

import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { Redirect, Route, RouteComponentProps, Switch } from 'react-router';

import { SUPPORT_LANGUAGES } from '../../../constants';
import AboutPage from '../../about/AboutPage';
import AccessbilityPage from '../../accessibility/AccessbilityPage';
import CollectionPageContainer from '../../collection/CollectionPageContainer';
import CollectionListPage from '../../collectionList/CollectionListPage';
import CoursePageContainer from '../../course/CoursePageContainer';
import CourseSearchPageContainer from '../../courseSearch/CourseSearchPageContainer';
import EventPageContainer from '../../event/EventPageContainer';
import EventSearchPageContainer from '../../eventSearch/EventSearchPageContainer';
import LandingPage from '../../landingPage/FrontPage';
import LandingPagePreview from '../../landingPage/LandingPagePreview';
import NotFound from '../../notFound/NotFound';
import { ROUTES } from './constants';

const App: FunctionComponent<RouteComponentProps<{
  locale: SUPPORT_LANGUAGES;
}>> = ({
  match: {
    params: { locale },
  },
}) => {
  const { i18n } = useTranslation();

  React.useEffect(() => {
    i18n.changeLanguage(locale);
  }, [i18n, locale]);

  return (
    <Switch>
      <Redirect exact path={`/${locale}/`} to={`/${locale}${ROUTES.HOME}`} />
      <Route exact path={`/${locale}${ROUTES.HOME}`} component={LandingPage} />
      <Route
        exact
        path={`/${locale}${ROUTES.HOME_PREVIEW}`}
        component={LandingPagePreview}
      />
      <Route exact path={`/${locale}${ROUTES.ABOUT}`} component={AboutPage} />
      <Route
        exact
        path={`/${locale}${ROUTES.ACCESSIBILITY}`}
        component={AccessbilityPage}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COLLECTIONS}`}
        component={CollectionListPage}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COLLECTION}`}
        component={CollectionPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.EVENTS}`}
        component={EventSearchPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.EVENT}`}
        component={EventPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.EVENT_PLACE}`}
        component={EventSearchPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COURSES}`}
        component={CourseSearchPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COURSE}`}
        component={CoursePageContainer}
      />
      {/* Redirect to next single event page url */}
      <Redirect
        exact
        path={`/${locale}${ROUTES.EVENT_DEPRECATED}`}
        to={`/${locale}${ROUTES.EVENT}`}
      />
      <Route component={NotFound} />
    </Switch>
  );
};

export default App;
