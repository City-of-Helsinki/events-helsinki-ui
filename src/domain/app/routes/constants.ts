import { MAPPED_PLACES } from '../../eventSearch/constants';

// Remember also update static urls to updateSitemap
export type ROUTE_TYPES = {
  ABOUT: '/about';
  ACCESSIBILITY: '/accessibility';
  COLLECTION: '/collection/:slug';
  COLLECTIONS: '/collections';
  EVENTS: '/events';
  EVENT: '/event/:id';
  COURSES: '/courses';
  COURSE: '/course/:id';
  HOME: '/home';
  HOME_PREVIEW: '/home/:id';
  EVENT_PLACE: string;
};

export const ROUTES: ROUTE_TYPES = {
  ABOUT: '/about',
  ACCESSIBILITY: '/accessibility',
  COLLECTION: '/collection/:slug',
  COLLECTIONS: '/collections',
  EVENTS: '/events',
  EVENT: '/event/:id',
  COURSES: '/courses',
  COURSE: '/course/:id',
  HOME: '/home',
  HOME_PREVIEW: '/home/:id',
  EVENT_PLACE: `/:place(${Object.keys(MAPPED_PLACES).join('|')})`,
};
