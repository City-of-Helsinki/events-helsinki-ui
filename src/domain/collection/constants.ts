import { mockEventData } from "../event/constants";

export const mockCollection = {
  collectionDetails: {
    curatedEvents: [
      `http://localhost:3000/fi/event/${mockEventData.eventDetails.id}`
    ],
    curatedEventsTitle: {
      en: "At least visit these",
      fi: "Käy ainakin näissä",
      sv: "Besök åtminstone dessa"
    },
    description: {
      en:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      fi:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      sv:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus."
    },
    eventListQuery:
      "http://localhost:3000/en/events?categories=music&districts=kaupunginosa%3Aetu-t%C3%B6%C3%B6l%C3%B6",
    eventListTitle: {
      en: "All the best events of the fall",
      fi: "Kaikki syksyn parhaat tapahtumat",
      sv: "Höstens bästa händelser"
    },
    id: "1",
    linkText: {
      en: "Read more on the project website",
      fi: "Lue lisää hankkeen omilta sivuilta",
      sv: "Läs mer på projektets webbplats"
    },
    linkUrl: {
      en: "http://www.google.com",
      fi: "http://www.google.com",
      sv: "http://www.google.com"
    },
    similarCollectionsTitle: {
      en: "Check also these",
      fi: "Katso myös nämä",
      sv: "Se dessa också"
    },
    socialMediaDescription: {
      en: "We put together the best foliage for the fall",
      fi: "Kokosimme parhaat tärpit syksylle",
      sv: "Vi sätter ihop det bästa bladverket för hösten"
    },
    subtitles: {
      en: "Subtitles en",
      fi: "Subtitles fi",
      sv: "Subtitles sv"
    },
    title: {
      en: "The most bustling tricks of the fall",
      fi: "Syksyn riehakkaimmat riennot",
      sv: "Höstens mest livliga knep"
    }
  }
};

// Amount of the similar collections
export const SIMILAR_COLLECTIONS_AMOUNT = 8;
