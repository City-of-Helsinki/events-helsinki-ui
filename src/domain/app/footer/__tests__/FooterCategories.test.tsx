import i18n from 'i18next';
import { axe } from 'jest-axe';
import React from 'react';

import { render, screen, userEvent } from '../../../../test/testUtils';
import { ROUTES } from '../../routes/constants';
import FooterCategories from '../FooterCategories';

beforeEach(() => {
  i18n.changeLanguage('fi');
});

test('component should be accessible', async () => {
  const { container } = render(<FooterCategories route={ROUTES.EVENTS} />);

  expect(await axe(container)).toHaveNoViolations();
});

test('should route to event search page by clicking category ', () => {
  const { history } = render(<FooterCategories route={ROUTES.EVENTS} />);

  userEvent.click(screen.getByRole('button', { name: /elokuva/i }));

  expect(history.location.pathname).toMatchSnapshot();
});

//  TODO: It seems that hds Footer does not support logoLanguage yet
test.skip('should show Swedish logo ', () => {
  i18n.changeLanguage('sv');
  render(<FooterCategories route={ROUTES.EVENTS} />);

  userEvent.click(screen.getByRole('button', { name: /elokuva/i }));

  expect(screen.queryByRole('img')).toHaveClass('sv');
});
