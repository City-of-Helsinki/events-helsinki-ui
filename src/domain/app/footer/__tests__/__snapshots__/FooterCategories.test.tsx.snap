// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should route to event search page by clicking category 1`] = `"/fi/events"`;
